raiden/raiden_service.py

```diff
@@ -696,271 +1132,323 @@ def on_message(self, message, msghash):  # noqa pylint: disable=unused-argument
-                    )
+    def handle_channelnew(self, state_change):
+        manager_address = state_change.manager_address
+        channel_address = state_change.channel_address
+        participant1 = state_change.participant1
+        participant2 = state_change.participant2
 
-        else:
-            log.info('ignoring new channel, this node is not a participant.')
-
-    def event_channelnewbalance(self, netting_contract_address_bin, event):
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug(
-                'channel new balance event received',
-                netting_contract=pex(netting_contract_address_bin),
-                event=event,
+        token_address = self.raiden.manager_token[manager_address]
+        graph = self.raiden.channelgraphs[token_address]
+        graph.add_path(participant1, participant2)
+
+        if participant1 == self.raiden.address or participant2 == self.raiden.address:
+            self.raiden.register_netting_channel(
+                token_address,
+                channel_address,
             )
 
-        token_address_bin = address_decoder(event['token_address'])
-        participant_address_bin = address_decoder(event['participant'])
+    def handle_balance(self, state_change):
+        channel_address = state_change.channel_address
+        token_address = state_change.token_address
+        participant_address = state_change.participant_address
+        balance = state_change.balance
+        block_number = state_change.block_number
 
-        # should not raise, all three addresses need to be registered
-        manager = self.raiden.get_manager_by_token_address(token_address_bin)
-        channel = manager.get_channel_by_contract_address(netting_contract_address_bin)
-        channel_state = channel.get_state_for(participant_address_bin)
+        graph = self.raiden.channelgraphs[token_address]
+        channel = graph.address_channel[channel_address]
+        channel_state = channel.get_state_for(participant_address)
 
-        if channel_state.contract_balance != event['balance']:
-            channel_state.update_contract_balance(event['balance'])
+        if channel_state.contract_balance != balance:
+            channel_state.update_contract_balance(balance)
 
         if channel.external_state.opened_block == 0:
-            channel.external_state.set_opened(event['block_number'])
-
-    def event_channelclosed(self, netting_contract_address_bin, event):
-        channel = self.raiden.find_channel_by_address(netting_contract_address_bin)
-        channel.external_state.set_closed(event['block_number'])
-
-    def event_channelsettled(self, netting_contract_address_bin, event):
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug(
-                'channel settle event received',
-                netting_contract=pex(netting_contract_address_bin),
-                event=event,
-            )
+            channel.external_state.set_opened(block_number)
```

Comment:
Yes, we don't have a smart contract call for "deploy the netting channel sub contract and deposit", we actually do first a "channel_manager.new_netting_channel" and then "netting_channel.deposit", for this reason a channel may be created but without funds for a period of time, so it's considered not open.